src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -987,22 +1014,121 @@ private static ConnectionMultiplexer CreateMultiplexer(object configuration, Log
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
         public static ConnectionMultiplexer Connect(string configuration, TextWriter log = null)
         {
-            SocketConnection.AssertDependencies();
-            return ConnectImpl(configuration, log);
+            var config = PrepareConfig(configuration);
+            return Connect(config, log);
         }
 
         /// <summary>
         /// Create a new ConnectionMultiplexer instance
         /// </summary>
-        /// <param name="configuration">The configurtion options to use for this multiplexer.</param>
+        /// <param name="configuration">The configuration options to use for this multiplexer.</param>
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
         public static ConnectionMultiplexer Connect(ConfigurationOptions configuration, TextWriter log = null)
         {
             SocketConnection.AssertDependencies();
-            return ConnectImpl(configuration, log);
+
+            bool sentinel = !String.IsNullOrEmpty(configuration.ServiceName);
+
+            if (!sentinel)
```

Comment:
Same as above, let's invert please - can eliminate the `sentinel` variable and put a comment here as well.